test/kubemark/pre-existing/README.md

```diff
@@ -0,0 +1,36 @@
+# Kubemark Pre-existing Provider Guide
+
+## Introduction
+
+The goal of the pre-existing provider is to land kubemark containers onto
+a pre-existing Kubernetes Cluster. It's meant to provide the developer with
+additional flexibility to customize the cluster infrastructure and still use
+the kubemark setup tools.  It is an **advanced** use case and not recommended
+for those who are not familiar with setting up a Kubernetes cluster and the
+infrastructe that the cluster sits on top of.
+
+## Requirements
+
+To use the pre-existing provider, the expectation is that there's a Kubernetes
+master that is rechable at $MASTER_IP. The machine that the Kubernetes master is
+on has to be ssh able from the host that's executing the kubemark scripts. And
+the user on that machine has to be 'kubernetes'.
+
+Requirement checklist:
+- Set MASTER_IP to ip address to the Kubernetes master
+- The host where you execute the kubemark scripts must be able to ssh to
+  kubernetes@$MASTER_IP
+
+## Example Configuration
+
+_test/kubemark/cloud-provider-config.sh_
+
+```
+CLOUD_PROVIDER="pre-existing"
+KUBEMARK_IMAGE_MAKE_TARGET="push"
+CONTAINER_REGISTRY=docker.io
+PROJECT="rthallisey"
+KUBEMARK_IMAGE_MAKE_TARGET="push"
+MASTER_IP="192.168.121.29"
+MASTER_IP_AND_PORT="${MASTER_IP}:6443"
```

Comment:
I'm still not sure if this port is required to be provided. @wojtek-t / @gmarek WDYT?